engines/aws/app/controllers/aws/credentials_controller.rb

```diff
@@ -2,7 +2,9 @@ module AWS
   class CredentialsController < ApplicationController
     before_action :load_credential, only: [:edit, :show]
 
-    def edit; end
+    def edit
+      redirect_to("/") unless ApplicationController.helpers.valid_login?
```

Comment:
Any reason you did this in every action instead of using a before_action filter in ApplicationController ?

Reply:
There is an overlap with the redirects, so in the `StepController`, the method that checks the access is `index` 
same as the method in `LoginController`, so `before_action` for `index` would make the `index` in `LoginController`
to not work (cause an error) 

I could do it for `edit` and `show` only and leave `StepController` check in `index` there 

and also I will debug the `SizeController` case because, besides being `edit`, the `before_action` is ignored

Reply:
Please see https://guides.rubyonrails.org/action_controller_overview.html#filters ; you can `skip_before_action` for the '/' action.